spinnaker_graph_front_end/examples/hello_world_untimed/hello_world_vertex.py

```diff
@@ -0,0 +1,133 @@
+# Copyright (c) 2017-2019 The University of Manchester
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from enum import Enum
+import logging
+from spinn_utilities.overrides import overrides
+from pacman.model.graphs.machine import MachineVertex
+from pacman.model.resources import ResourceContainer, VariableSDRAM
+from spinn_front_end_common.utilities.constants import (
+    SYSTEM_BYTES_REQUIREMENT, BYTES_PER_WORD)
+from spinn_front_end_common.utilities.helpful_functions import (
+    locate_memory_region_for_placement)
+from spinn_front_end_common.abstract_models import (
+    AbstractGeneratesDataSpecification)
+from spinn_front_end_common.interface.buffer_management.buffer_models import (
+    AbstractReceiveBuffersToHost)
+from spinn_front_end_common.interface.buffer_management import (
+    recording_utilities)
+from spinnaker_graph_front_end.utilities import SimulatorVertex
+from spinnaker_graph_front_end.utilities.data_utils import (
+    generate_system_data_region)
+import numpy
+from pacman.executor.injection_decorator import inject_items
+
+logger = logging.getLogger(__name__)
+
+
+class HelloWorldVertex(
+        SimulatorVertex, AbstractGeneratesDataSpecification,
```

Comment:
so its a simulator vertex is it? dont look like a simulator vertex to me. so your c code is going to have a runtime handed to it, which it isnt going to follow.............. seems a bit of a fudge here. 

why not have a extra level of inheretance, and allow the chip runtime updator to only look for the timed ones?

Reply:
It is a simulator vertex.  The idea is very similar to auto pause and resume or run-and-run-again.  The only difference is that it runs in steps instead of time steps.  So the interface works just the same.